pkg/internal/computeresources/transformer.go

```diff
@@ -99,24 +99,38 @@ func transformPodBasedOnLimitRange(p *corev1.Pod, limitRange *corev1.LimitRange)
 			p.Spec.Containers[i].Resources.Requests = defaultRequests
 		} else {
 			for _, name := range resourceNames {
-				setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Requests, defaultRequests)
+				setRequests(name, p.Spec.Containers[i].Resources.Requests, defaultRequests, defaultContainerLimits)
 			}
 		}
 		if p.Spec.Containers[i].Resources.Limits == nil {
 			p.Spec.Containers[i].Resources.Limits = defaultContainerLimits
 		} else {
 			for _, name := range resourceNames {
-				setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Limits, defaultContainerLimits)
+				setLimits(name, p.Spec.Containers[i].Resources.Limits, defaultContainerLimits)
 			}
 		}
 	}
 	return p
 }
 
-func setRequestsOrLimits(name corev1.ResourceName, dst, src corev1.ResourceList) {
-	if limitrange.IsZero(dst[name]) && !limitrange.IsZero(src[name]) {
-		dst[name] = src[name]
+// setRequests sets the requests to the max of (currentRequests, defaultRequests)
+// and the min of (currentRequests, defaultLimits)
+func setRequests(name corev1.ResourceName, dst corev1.ResourceList, requests corev1.ResourceList, limits corev1.ResourceList) {
```

Comment:
think some discussion will come to this line:

my current understanding as 
- if not limitranges, just stay as it is

- if currentRequest < min (of limitrange), keep the min
- if over the limit, set as the limit

for limits
- similarly, but keep the min of currentLimit and limit from limitrange